string/string_like.mbt

```diff
@@ -26,3 +26,9 @@ pub impl ToStringView for String with to_string_view(self) -> StringView {
 pub impl ToStringView for StringView with to_string_view(self) -> StringView {
   self
 }
+
+///|
+/// Implements ToStringView for Char, enabling operations like Array::join on Array[Char]
+pub impl ToStringView for Char with to_string_view(self) -> StringView {
+  self.to_string()[:]
```

Comment:
```suggestion
  [self]
```

In MoonBit, we can use array syntax to construct String by Char. And there's implicit conversion from String to StringView